### ISSUE TITLE
Module not found: Error: Can't resolve 'fs'

### ISSUE THREAD

**Issue by T00rk** [rohit-gohri/redocusaurus#57]
I've updated my docusaurus project to `2.0.0-beta.0` and redocusaurus is not working anymore due to this error 

`Module not found: Error: Can't resolve 'fs'`

```sh
Module not found: Error: Can't resolve 'fs' in '/home/romain/Projets/help/node_modules/redoc/bundles'
Module not found: Error: Can't resolve 'process/browser' in '/home/romain/Projets/help/node_modules/yaml/browser/dist'
Did you mean 'browser.js'?
BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
```

**Comment by rohit-gohri** [comment]
Can you also update redocusaurus to v0.4.2? In the beta webpack was upgraded to v5 that caused build issues.

Related to #25

**Comment by T00rk** [comment]
The last available version on npm is 0.4.1

**Comment by MaxandreJ** [comment]
Hi, thanks for developing this package! Is there a way to make it work? I'm a novice with `npm` and I'm not sure how to install a specific version of your package from github.

I tried
``` npm install https://github.com/rohit-gohri/redocusaurus#4d415d68d3f07cbbd9757f39ba68f915586379e4 --save redocusaurus ```

(`4d415d68d3f07cbbd9757f39ba68f915586379e4` is your v0.4.2 commit)

but I still have the same issue

**Comment by rohit-gohri** [comment]
Yeah, sorry. 0.4.1 is the latest. 0.4.2 was for the internal package. I'm not sure what the problem is here because I have it running with 2.0.0-beta in the examples: https://github.com/rohit-gohri/redocusaurus/tree/main/example

Could one of you share a minimal repo where this is happening?

**Comment by T00rk** [comment]
I've tried making the example working and it doesn't.

Le lun. 14 juin 2021 à 13:42, Rohit Gohri ***@***.***> a
écrit :

**Comment by patrick-simonson** [comment]
+1 - Cloning your repo and doing an npm install + npm run start in the example folder brings up the issue for me.

**Comment by drichards-87** [comment]
I was encountering the same issue (see my comment in #14). I was able to workaround it by downgrading the installed version of Redoc to v2.0.0-rc.53.

**Comment by rohit-gohri** [comment]
I think I would have to pin redoc dependency to a specific version instead of using `^`

**Comment by bash83** [comment]
Same issue using GitHub Actions to compile and publish the docs to GitHub Pages:

`
Module not found: Error: Can't resolve 'fs' in '/home/runner/work/docs/docs/node_modules/redoc/bundles'
`

**Comment by sibelius** [comment]
create a `sitePlugin.js` file, so you can customize docusuarus webpack

```jsx
module.exports = function (context, options) {
  return {
    name: 'custom-docusaurus-plugin',
    // eslint-disable-next-line
    configureWebpack(config, isServer, utils) {
      return {
        resolve: {
          alias: {
            path: require.resolve('path-browserify'),
          },
          fallback: {
            fs: false,
            http: require.resolve('stream-http'),
            https: require.resolve('https-browserify'),
            os: require.resolve('os-browserify/browser'),
          },
        },        
      };
    },
  };
};
```

import on docusuarus.config.js

```jsx
module.exports = {
plugins: [require.resolve('./sitePlugin')],
}
```

**Comment by rohit-gohri** [comment]
@sibelius That looks promising, could you open a PR adding them here: https://github.com/rohit-gohri/redocusaurus/blob/main/packages/docusaurus-theme-redoc/src/index.ts#L18-L27

**Comment by rohit-gohri** [comment]
Should be fixed in 0.4.4
